client/additional-methods-setup/upe-preview-methods-selector/index.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Card, CardBody } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import Wizard from '../wizard/wrapper';
+import WizardTask from '../wizard/task';
+import WizardTaskList from '../wizard/task-list';
+import EnableUpePreviewTask from './enable-upe-preview-task';
+import SetupCompleteTask from './setup-complete-task';
+import './index.scss';
+import useIsUpeEnabled from '../../settings/wcpay-upe-toggle/hook';
+import WcpaySettingsFetcher from './wcpay-settings-fetcher';
+
+const UpePreviewMethodsSelector = () => {
+	const [ isUpeEnabled ] = useIsUpeEnabled();
+
+	return (
+		<Card className="upe-preview-methods-selector">
+			<CardBody>
+				<Wizard
+					defaultActiveTask={
+						isUpeEnabled ? 'setup-complete' : 'enable-upe-preview'
+					}
+					defaultCompletedTasks={ {
+						'enable-upe-preview': isUpeEnabled,
+					} }
+				>
+					<WizardTaskList>
+						<WizardTask id="enable-upe-preview">
+							<EnableUpePreviewTask />
+							<WcpaySettingsFetcher />
```

Comment:
I like to use components that can act as units of work.
> Components let you split the UI into independent, reusable pieces, and think about each piece in isolation - https://reactjs.org/docs/components-and-props.html

In their purest form, components act as individual pieces of the UI. As the application grows, components also evolve and tend to encompass a lot of other functionality/responsibilities: add event listeners, fetch data, handle navigation, manage state, track user interactions, etc.
We use the React lifecycle to control the execution flow of our application between renders. Components don't need to render anything, but they can still _do_ something. Returning `null` is valid, and it can be a powerful pattern, which we can use to "componentize" the lifecycle logic. Which can also make things becoming pretty declarative, too!
Like:
```jsx
if(!isLoggedIn) return <RedirectTo url={loginUrl} />
```

Or even in `react-router`, also very declarative:
```jsx
  <Router>
    <Route path="/home" render={() => isLoggedIn ? <Dashboard /> : <Redirect to="/login" /> } />
  </Router>
```
Or even this interesting component in `react-router` (that I guess now is kinda outdated, since we can just use hooks in a similar way, like here 😄 ) that uses props to execute stuff: https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/Lifecycle.js

So again, I think that placing execution logic in components is a powerful pattern - afterall components are just functions that can render (aka return something different than `null`) or do something (through `useEffect`, for example).

Let me know if I went too much on a tangent! 😅

Reply:
Thanks for the awesome explanation - I really appreciate it whenever I ask about something and you post a comment like this one! 🙌 

I guess I just always treat React components as building blocks of a UI and if they don't do UI-related stuff, my intuition is not to use a React component and put stuff in a method/function/a non-React object. On second thought - contexts aren't exactly known for rendering too much either! 😄 

Thanks for describing this approach! Now that I understand the motivation, I really like it! 👍